aws/resource_aws_servicecatalog_product.go

```diff
@@ -0,0 +1,450 @@
+			},
+			"distributor": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"product_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"has_default_path": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"product_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"support_description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"support_email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"support_url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"provisioning_artifact": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"info": {
+							Type:     schema.TypeMap,
+							Required: true,
+							Elem:     schema.TypeString,
+							ForceNew: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Default:  servicecatalog.ProvisioningArtifactTypeCloudFormationTemplate,
+						}, // CLOUD_FORMATION_TEMPLATE  | MARKETPLACE_AMI | MARKETPLACE_CAR
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"active": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"created_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsServiceCatalogProductCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+	input := servicecatalog.CreateProductInput{}
+
+	if v, ok := d.GetOk("name"); ok {
+		input.Name = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("distributor"); ok {
+		input.Distributor = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		input.Name = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("owner"); ok {
+		input.Owner = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("product_type"); ok {
+		input.ProductType = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("support_description"); ok {
+		input.SupportDescription = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("support_email"); ok {
+		input.SupportEmail = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("support_url"); ok {
+		input.SupportUrl = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		input.Tags = tagsFromMapServiceCatalog(v.(map[string]interface{}))
```

Comment:
use the kayvaluetags func here.
`keyvaluetags.New(v.(map[string]interface{})).IgnoreAws().ServicecatalogTags()`

Reply:
👍